### DISCUSSION TITLE
Failed substitution: echo "hello" > foo && cat $(ls foo)

### DISCUSSION THREAD

**Discussion by nbelakovski** [xonsh/xonsh#5035] [Q&A] [answered]
I'm just getting started with xonsh. I love the idea of a shell with basic arithmetic. Seems like a basic thing that every shell should have out of the box, but anyway.... I'm trying to learn how it works and I've run into an error with the following toy example. What's the correct way to do this?

```
>>> echo "hello" > foo && cat foo
hello
>>> echo "hello" > foo && cat $(ls foo)
cat: 'foo'$'\n': No such file or directory
```

This seems like a pretty straightforward issue, are there changes that can be made to Xonsh to make this work by default? Seems like it would make it easier for people to migrate from bash and other shells.

**Comment by anki-code** [comment, marked as answer]
First of all start from [хonsh-cheatsheet](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md) - the best doc for new users.

The issue you described is in the section [The process substitution operator $() returns output with universal new lines](https://github.com/anki-code/xonsh-cheatsheet/blob/main/README.md#3-the-process-substitution-operator--returns-output-with-universal-new-lines).
